src/frontend/src/components/modal.ts

```diff
@@ -7,78 +7,47 @@ import { withRef } from "../utils/lit-html";
  **/
 export const createModal = ({
   slot,
-  closeButtonRef,
-  submitButtonRef,
+  submit,
 }: {
   slot: TemplateResult;
-  closeButtonRef: Ref<HTMLButtonElement>;
-  submitButtonRef: Ref<HTMLButtonElement>;
-}): HTMLDialogElement => {
+  submit: () => void;
+}) => {
+  const modalElement: Ref<HTMLDialogElement> = createRef();
+
+  const closeModal = () =>
+    withRef(modalElement, (modalElement) => {
+      modalElement.close();
+      modalElement.remove();
```

Comment:
this means that clicking the "X" button removes the element, but submitting doesn't. Will need to figure out what we want exactly.